pilot/pkg/model/sidecar.go

```diff
@@ -41,7 +41,11 @@ var (
 	}
 
 	sidecarScopeNamespaceConfigTypes = map[resource.GroupVersionKind]struct{}{
-		gvk.Sidecar: {},
+		gvk.Sidecar:               {},
+		gvk.EnvoyFilter:           {},
+		gvk.AuthorizationPolicy:   {},
+		gvk.PeerAuthentication:    {},
```

Comment:
Looks good for RequestAuthentication and AuthorizationPolicy, but not sure if there is any implications for autoMTLS with PeerAuthentication, @incfly can you confirm. thanks.

Reply:
PeerAuthentication policy for server side only affect its own namespace all entire mesh if the CR is in root namespace. 

But if `foo` ns changed from STRICT to plaintext, we expect the client side's auto mTLS config in bar namespace to respect that as well, should receive CDS push for the update to remove the auto mTLS config.

Will this PR break above requirement? @howardjohn